econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -0,0 +1,1056 @@
+        model, then only the features in this array will be used for heterogeneity. If a 2d
+        array then its first dimension should be len(feature_inds) and whenever estimating
+        a local causal effect for target feature feature_inds[i], then only features in
+        heterogeneity_inds[i] will be used for heterogeneity. If heterogeneity_inds[i]=None, then all features
+        are used for heterogeneity when estimating local causal effect for feature_inds[i]. If heterogeneity_ind=None
+        then all features are used for heterogeneity for all features.
+    feature_names: list of str, default None
+        The names for all of the features in the data.  Not necessary if the input will be a dataframe.
+        If None and the input is a plain numpy array, generated feature names will be ['X1', 'X2', ...].
+    upper_bound_on_cat_expansion: int, default 5
+        The maximum number of categorical values allowed, because they are expanded via one-hot encoding. If a
+        feature has more than this many values, then a causal effect model is not fitted for that target feature
+        and a warning flag is raised. The remainder of the models are fitted.
+    classification: bool, default False
+        Whether this is a classification (as opposed to regression) task
+        TODO. Enable also multi-class classification (post-MVP)
+    nuisance_models: one of {'linear', 'automl'}, optional (default='linear')
+        What models to use for nuisance estimation (i.e. for estimating propensity models or models of how
+        controls predict the outcome). If 'linear', then LassoCV (for regression) and LogisticRegressionCV
+        (for classification) are used. If 'automl', then a kfold cross-validation and model selection is performed
+        among several models and the best is chosen.
+        TODO. Add other options, such as {'azure_automl', 'forests', 'boosting'} that will use particular sub-cases
+        of models or also integrate with azure autoML. (post-MVP)
+    heterogeneity_model: one of {'linear', 'forest'}, optional (default='linear')
+        What type of model to use for treatment effect heterogeneity. 'linear' means that a heterogeneity model
+        of the form theta(X)=<a, X> will be used, while 'forest' means that a forest model will be trained instead.
+        TODO. Add other options, such as {'automl'} for performing
+        model selection for the causal effect, or {'sparse_linear'} for using a debiased lasso. (post-MVP)
+    automl: bool, default True
+        Whether to automatically perform model selection over a variety of models
+    n_jobs: int, default -1
+        Degree of parallelism to use when training models via joblib.Parallel
+    """
+
+    _results = namedtuple("_results", field_names=[
+        "feature_index", "feature_name", "feature_baseline", "feature_levels", "hinds",
+        "X_transformer", "W_transformer", "estimator", "global_inference", "d_t"])
```

Comment:
We might want to remove d_t if `_results.d_t` not used anywhere

Reply:
Done